### PR TITLE
added reason text box in archiving process

### PR DESCRIPTION
**Pull request by gauravsinhaweb**
**Issue - #550**
### What is the change?

Added Reason text box in the archiving process

### Is it bug?

- Steps to repro
- Expected
- Actualo

### \*Behind Feature flag?

- [x] Yes
- [ ] No

### \*Dev Tested?

- [x] Yes
- [ ] No

#### Platforms

- [ ] Web

#### Browsers

- [ ] Chrome
- [ ] Safari
- [ ] Firefox

### Before / After Change Screenshots

https://github.com/Real-Dev-Squad/website-members/assets/75125943/5b90d0b9-ff5a-4d58-a40b-92f1c9455da9

Screenshot 2023-09-14 015427
Screenshot 2023-09-14 015253

### CODE REVIEW

File: src/components/member-role-update/member-role-update.module.scss [hunk 1]
```diff
@@ -9,3 +9,24 @@
...
+  }
+  &__error {
+    color: #e13110;
+    text-align: center;
+    font-size: inherit;
+  }
+  &__success {
+    color: #1e7e34;
+    text-align: center;
+    font-size: inherit;
```
Comment (Pratiyushkumar) [inline src/components/member-role-update/member-role-update.module.scss:30]: please use color variable names instead of color codes directly

**Review by Pratiyushkumar — CHANGES_REQUESTED** [review]

File: src/components/member-role-update/index.js [hunk 2]
```diff
@@ -6,10 +6,14 @@ import Spinner from '@components/UI/spinner';
 import MemberTagAssign from '@components/member-tag-assign';
 import { BASE_API_URL } from '@constants/AppConstants';
 import useFetch from '@custom-hooks/useFetch';
+import { useRouter } from 'next/router';
 import classNames from './member-role-update.module.scss';
 import { memberRoleUpdate } from '../../helper-functions/action-handlers';
 
 const MemberRoleUpdate = () => {
+  const { query } = useRouter() || { query: { dev: false } };
+  const { dev } = query;
```
[the remark this answers is not in the document]
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:15]: both ways will function the same, it's the developer's choice

File: src/components/member-role-update/index.js [hunk 3]
```diff
@@ -6,10 +6,14 @@ import Spinner from '@components/UI/spinner';
...
+  const isDev = Boolean(dev); // convert string to boolean
```
[the remark this answers is not in the document]
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:16]: tell me why I should use this way

File: src/components/member-role-update/index.js [hunk 4]
```diff
@@ -51,22 +56,31 @@ const MemberRoleUpdate = () => {
     }
   };
 
-  const archiveUnArchiveTheMember = async (id) => {
-    let archiveRole = null;
+  const archiveUnArchiveTheMember = async (id, reason) => {
     setIsUpdating(true);
+    let body = {};
```
[the remark this answers is not in the document]
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:61]: because it is supposed to pass as a body in the archiving API, Let me know if you have a better name

File: src/components/member-role-update/index.js [hunk 5]
```diff
@@ -75,30 +89,86 @@ const MemberRoleUpdate = () => {
...
     >
       {archived ? 'Unarchive Member' : 'Archive Member'}
     </button>
   );
 
+  const archiveReasonTextBox = (
+    <div className={classNames.archiveUser}>
+      <p className={classNames.archiveUser__error}>{validateError}</p>
+      <label htmlFor="archiveReason" data-testid="reasonInputLabel">
+        Reason:{' '}
```
[the remark this answers is not in the document]
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:149]: updated

File: src/components/member-role-update/member-role-update.module.scss (hunk 1 above)
Comment (gauravsinhaweb) [reply src/components/member-role-update/member-role-update.module.scss:30]: updated

File: src/components/member-role-update/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:16]: You're correct. If we don't need to worry about type coercion and only want to check if the dev property is a string "true," we can perform a conditional check. But what if we encounter values like null, undefined, or even the boolean true,
the code will proceed to the else block. The `|| { query: { dev: false } }` part was added to handle these non-string values,
Let me know if you still want me to remove `|| { query: { dev: false } }`, with a reason

File: src/components/member-role-update/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:15]: I have addressed below

File: src/components/member-role-update/index.js [hunk 6]
```diff
@@ -51,22 +56,31 @@ const MemberRoleUpdate = () => {
...
+        archived: false,
+      };
+    } else if (!reason && !archived) {
+      // if user is not archived and reason is not provided
+      body = {
+        archived: true,
+      };
     } else {
-      archiveRole = true;
+      // if user is not archived and reason is provided
```
Comment (sahsisunny) [inline src/components/member-role-update/index.js:73]: These comments are important?

File: src/components/member-role-update/index.js [hunk 7]
```diff
@@ -75,30 +89,86 @@ const MemberRoleUpdate = () => {
...
+      default:
+        setvalidateError('');
+    }
+    const isValid =
+      !isEmptyReason &&
+      !isReasonEmptyOrWhitespace &&
+      !isMoreThan99Words &&
+      !isMoreThan50Characters;
+    if (isValid || archived) {
+      archiveUnArchiveTheMember(id, reason); // if valid reason or unarchiving, then archive/unarchive
```
Comment (sahsisunny) [inline src/components/member-role-update/index.js:127]: Here also. There is no need to put comments for logic you implemented.

Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:127]: updated

File: src/components/member-role-update/index.js (hunk 6 above)
Comment (gauravsinhaweb) [reply src/components/member-role-update/index.js:73]: no, updated

File: .env.development
Comment (Pratiyushkumar) [inline .env.development:1]: why this change??

**Review by Pratiyushkumar — CHANGES_REQUESTED** [review]

File: .env.development
Comment (gauravsinhaweb) [reply .env.development:1]: fixed

**Review by Pratiyushkumar — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]
